File: src/main/java/edu/wayne/pfsdm/FieldFeature.java
package edu.wayne.pfsdm;

import org.lemurproject.galago.core.retrieval.Retrieval;

import java.util.List;

/**
 * Created by fsqcds on 5/1/15.
 */
public abstract class FieldFeature {
    protected ParametrizedFSDMTraversal traversal;
    protected String fieldName;
    public FieldFeature(String fieldName, ParametrizedFSDMTraversal traversal) {
        this.fieldName = fieldName;
        this.traversal = traversal;
    }
    public abstract double getPhi(List<String> tokens);
}

File: src/main/scala/edu/wayne/pfsdm/FieldFeature.scala
package edu.wayne.pfsdm

import edu.wayne.pfsdm.feature.{CapitalizedFeature, BaselineTopScoreFieldFeature}

import scala.collection.JavaConversions._

/**
 * Created by fsqcds on 5/1/15.
 */
trait FieldFeature {
  /**
   * Returns feature value for field and tokens.
   *
   * @param tokens    terms for which feature is calculated
   * @param fieldName name of the field. This is ignored for features that doesn't depend on field
   * @return          feature value
   */
  def getPhi(tokens: Seq[String], fieldName: String): Double

  /**
   * Returns feature value for field and tokens. This is Java wrapper for getPhi(Seq[String], String).
   *
   * @param tokens    terms for which feature is calculated
   * @param fieldName name of the field. This is ignored for features that doesn't depend on field
   * @return          feature value
   */
  def getPhi(tokens: java.lang.Iterable[String], fieldName: String): Double = getPhi(tokens.toList, fieldName: String)
}

object FieldFeature {
  def apply(fieldFeatureName: String, traversal: ParametrizedFSDMTraversal) = fieldFeatureName match {
    case "baselinetopscore" => new BaselineTopScoreFieldFeature(traversal)
    case "capitalized" => new CapitalizedFeature()
  }
}